lib/foreman_ansible_core/playbook_runner.rb

```diff
@@ -94,5 +97,41 @@ def initialize_ansible_dir(ansible_dir)
         !ansible_dir.nil? && File.exist?(ansible_dir)
       @ansible_dir = ansible_dir
     end
+
+    def found_in_known_hosts?
+      JSON.parse(@inventory)['all']['hosts'].each do |host|
+        return false if Net::SSH::KnownHosts.search_for(host).empty?
+      end
+      true
+    end
+
+    def add_to_known_hosts
+      logger.warn('[foreman_ansible] - Host not found in known_hosts')
+      JSON.parse(@inventory)['all']['hosts'].each do |host|
+        stdout, stderr, status = Open3.capture3("ssh-keyscan #{host}")
+        if status.success?
+          # Add key to known_hosts
+          host_keys = stdout.split("\n")
+          Net::SSH::KnownHosts.hostfiles({}).each do |known_hosts_file|
+            # '~' cannot be properly substituted in Ruby as there's no shell
+            known_hosts_file.sub!('~', Dir.home)
```

Comment:
Why not `File.expand_path`?

Reply:
What's the difference?

Reply:
Perhaps it's nothing though and I'm being overly cautious. I think the File module is generally preferred when handling paths. I don't know if there's a difference, but I've learned that using the existing tools often avoids unexpected behavior in the long run. Remember that filenames can have `~` in them.

Reply:
Makes sense, changing it